src/Transform.cpp

```diff
@@ -141,14 +142,29 @@ void Transform::compute_model_matrix(glm::mat4 const& parent_global_model_matrix
     m_parent_dirty = false;
 }
 
+void Transform::set_model_matrix(glm::mat4 const& matrix)
+{
+    m_model_matrix = matrix;
+    if (parent.expired())
+    {
+        glm::decompose(m_model_matrix, m_local_scale, m_local_rotation, m_local_position, m_skew, m_perpective);
+        m_euler_angles = glm::degrees(glm::eulerAngles(m_local_rotation));
+    }
+    else
+    {
+        glm::decompose(glm::inverse(parent.lock()->get_model_matrix()) * m_model_matrix, m_local_scale, m_local_rotation, m_local_position, m_skew, m_perpective);
+        m_euler_angles = glm::degrees(glm::eulerAngles(m_rotation));
```

Comment:
Using `m_rotation` here seems pretty sus, gizmos work correctly, but euler angles shown in inspector are probably wrong.